#!/bin/bash
#PBS -l vmem=16GB
#PBS -l walltime=24:00:00
#PBS -N mut_location
#PBS -o /mount/samee/zianl_overflow/motif_seqcon/
#PBS -e /mount/samee/zianl_overflow/motif_seqcon/

cd $PBS_O_WORKDIR
ls

# Modules
module load bedtools/2.29.2

# Environmental variables
InMut="1kg_afr_muts_all.bed"
InBEDList="motifbeds/*.bed"
Genome=/mount/samee/Genomes/hg19_ucsc/hg19.fa
OutDir="motifmutlocs/"

echo "Input mutation BED file: "$InMut
echo "Reference genome: "$Genome
echo "Input TFBS BED file directory: "$InBEDList
echo "Output directory: "$OutDir


# Make a trimmed mutation file reference
cut -f1-4 $InMut > ___scratchin_1.bed

# Main loop
for InBED in $(ls $InBEDList); do
  # Output name
  Out=$(echo ${InBED##*/} | sed "s/motif/mutdist/")
  # Trim the inputs
  cut -f1-3,6 $InBED > ___scratchin_2.bed
  # Make BED return; 1-3: BED coordinates, 4: mutation, 5: MAF, 6: TFBS location, 7: k-mer
  bedtools intersect -a ___scratchin_1.bed -b ___scratchin_2.bed -wb |\
    awk 'OFS="\t" {
      if ($8=="+") {print $1,$2-3,$3+3,$4,$2-$6}
      else if ($8=="-") {print $1,$2-3,$3+3,$4,$7-$3}
    }' |\
    bedtools getfasta -fi $Genome -bed stdin -bedOut |\
    sed 's/_/\t/g' > ___scratchin_3.bed
    # Add 5-mers
    cut -f7 ___scratchin_3.bed | cut -c2-6 | paste ___scratchin_3.bed /dev/stdin | sed 's/ /\t/g' > $OutDir$Out

  # Remove scratch
  rm ___scratchin_2.bed
  rm ___scratchin_3.bed

done

# Remove scratch
rm ___scratchin_1.bed
